### PR TITLE
Configuration for `docs-builder`

### PR DESCRIPTION
**Pull request by pomek**
Fixes: #243 

This PR is strongly connected with https://github.com/cksource/docs-builder/pull/6

### CODE REVIEW

File: gulpfile.js [hunk 1]
```diff
@@ -130,6 +133,21 @@ gulp.task( 'bundle:generate',
 
 // Documentation. -------------------------------------------------------------
 
-const docsBuilder = ckeditor5DevCompiler.docs( config );
+// Documentation. -------------------------------------------------------------
+
+const ckeditor5DevDocs = require( '@ckeditor/ckeditor5-dev-docs' );
+const docsBuilder = ckeditor5DevDocs.docs( config );
+
+gulp.task( 'docs', [ 'compile:js:esnext' ], ( done ) => {
```
Comment (Reinmar) [inline gulpfile.js]: To avoid the same mistake as we did with the other tasks, we should work already on minimising the number of sub tasks which we need to create.

So, the ckeditor5-dev-docs should expose a single method which will make for the `docs` task. All the rest (the sub tasks) can still be exposed by that package (so if anyone will need them, they can use it) but they should not be necessary to register here. They need to be used internally by the function which makes for the `docs` task.

Comment (Reinmar) [inline gulpfile.js]: To add to this – most likely, we'll need a function which is able to take functions which generates streams (sub tasks) and make one big stream out of that. Gulp does this internally and `runSequence` uses that, but we need to bypass gulp. Otherwise, we'll get crazy.

File: package.json [hunk 2]
```diff
@@ -25,18 +25,19 @@
...
-    "@ckeditor/ckeditor5-dev-lint": "^1.0.1",
-    "@ckeditor/ckeditor5-dev-utils": "ckeditor/ckeditor5-dev-utils",
     "babel-polyfill": "^6.13.0",
     "benderjs": "^0.4.1",
     "benderjs-chai": "^0.2.0",
     "benderjs-coverage": "benderjs/benderjs-coverage#t/4",
     "benderjs-mocha": "^0.3.0",
     "benderjs-promise": "^0.1.0",
     "benderjs-sinon": "^0.3.0",
+    "@ckeditor/ckeditor5-dev-bundler-rollup": "ckeditor/ckeditor5-dev-bundler-rollup",
```
Comment (Reinmar) [inline package.json]: Why did you move all deps here? As far as I noticed, npm automatically sorts them so those starting with `@` are at the beginning (what makes sense).

File: docs/index.pug [hunk 3]
```diff
@@ -0,0 +1,11 @@
+extends /layout/base.pug
+
+block content
+	main.col-md-24
+		h3 CKEditor 5 Documentation
+		hr
```
Comment (Reinmar) [inline docs/index.pug]: This `<hr>` is absolutely incorrect here. If you want a line below a heading, then use CSS.

File: docs/index.pug [hunk 4]
```diff
@@ -0,0 +1,11 @@
+extends /layout/base.pug
```
Comment (Reinmar) [inline docs/index.pug]: I wonder whether it wouldn't be better if this was markdown. All the rest of the docs will be compiled from MD files, so it will look bad if we have one PUG file here. OTOH, PUG gives us more freedom for the index file, but I'd still go with MD. Any problems with that?

Comment (Reinmar) [inline docs/index.pug]: BTW. In MD we can also use HTML, so we shouldn't be that limited.

Comment (pomek) [inline docs/index.pug]: > Any problems with that?

No. We can use MD file as the main page. But we need to prepare a separate [template](https://github.com/cksource/docs-builder/tree/t/1/template) for the page. But, it isn't a problem.

File: package.json (hunk 2 above)
Comment (pomek) [inline package.json]: Probably I did it when I was resolving the git conflicts. I will fix it.

File: gulpfile.js [hunk 5]
```diff
@@ -17,14 +17,17 @@ const config = {
...
-	DOCS_DIR: 'build/docs',
 	BUNDLE_DIR: 'build/dist',
 	WORKSPACE_DIR: '..',
 
 	// Path to the default configuration file for bundler.
 	BUNDLE_DEFAULT_CONFIG: 'dev/bundles/build-config-standard.js',
 
 	DOCUMENTATION: {
+		// Path to the temporary documentation files.
+		SOURCE_DIR: '.docs',
```
Comment (Reinmar) [inline gulpfile.js]: I don't like this. The comment says something different than the name of this property. And do we have to keep some temporary stuff in the root dir? Can't we keep it somewhere in `build/`?

File: gulpfile.js (hunk 1 above)
Comment (pomek) [inline gulpfile.js]: Firstly we need to create a custom `runSequence` module for that. We have a ticket for this issue - https://github.com/ckeditor/ckeditor5-dev-utils/issues/21.

When the module will be ready, we can remove all sub tasks. At this moment let's leave it as it is.

File: docs/index.pug (hunk 3 above)
Comment (pomek) [inline docs/index.pug]: I will remove this file. We can use [`readme.md`](https://github.com/ckeditor/ckeditor5/blob/master/README.md) file as the main page for the documentation. In the future, we will be able to change it.

**Comment by pomek** [comment]
Ok. I introduced few changes but `docs-builder` needs to be adjusted to the changes. I will let you know when everything will be prepared.

**Comment by Reinmar** [comment]
When I tried to run it together I got:

```
[17:14:17] Compiling styles...
[17:14:17] An error occurs during the build process.
[17:14:17] 'docs:build' errored after 342 ms
[17:14:17] Error: File to import not found or unreadable: /www/ckeditor5/ckeditor5/node_modules/@ckeditor/ckeditor5-dev-docs/node_modules/docs-builder/node_modules/bootstrap-sass/assets/stylesheets/bootstrap
Parent style sheet: stdin
    at options.error (/www/ckeditor5/ckeditor5/node_modules/@ckeditor/ckeditor5-dev-docs/node_modules/node-sass/lib/index.js:292:32)
[17:14:17] 'docs' errored after 2.53 s
[17:14:17] Error in plugin 'run-sequence(docs:build)'
Message:
    File to import not found or unreadable: /www/ckeditor5/ckeditor5/node_modules/@ckeditor/ckeditor5-dev-docs/node_modules/docs-builder/node_modules/bootstrap-sass/assets/stylesheets/bootstrap
Parent style sheet: stdin
Details:
    formatted: Error: File to import not found or unreadable: /www/ckeditor5/ckeditor5/node_modules/@ckeditor/ckeditor5-dev-docs/node_modules/docs-builder/node_modules/bootstrap-sass/assets/stylesheets/bootstrap
       Parent style sheet: stdin
        on line 2 of stdin
>> @import '/www/ckeditor5/ckeditor5/node_modules/@ckeditor/ckeditor5-dev-docs/no
   ^

    column: 1
    line: 2
    file: stdin
    status: 1
Stack:
Error: File to import not found or unreadable: /www/ckeditor5/ckeditor5/node_modules/@ckeditor/ckeditor5-dev-docs/node_modules/docs-builder/node_modules/bootstrap-sass/assets/stylesheets/bootstrap
Parent style sheet: stdin
    at options.error (/www/ckeditor5/ckeditor5/node_modules/@ckeditor/ckeditor5-dev-docs/node_modules/node-sass/lib/index.js:292:32)
(t/243 daae923) p@m /www/ckeditor5/ckeditor5>
```

**Comment by pomek** [comment]
Please update the `docs-builder`. Most probably, I've fixed your issue.

**Comment by Reinmar** [comment]
Still the same:

```
[10:52:20] Error: File to import not found or unreadable: /www/ckeditor5/ckeditor5/node_modules/@ckeditor/ckeditor5-dev-docs/node_modules/docs-builder/node_modules/bootstrap-sass/assets/stylesheets/bootstrap
Parent style sheet: stdin
    at options.error (/www/ckeditor5/ckeditor5/node_modules/@ckeditor/ckeditor5-dev-docs/node_modules/node-sass/lib/index.js:292:32)
[10:52:20] 'docs' errored after 2.62 s
[10:52:20] Error in plugin 'run-sequence(docs:build)'
Message:
    File to import not found or unreadable: /www/ckeditor5/ckeditor5/node_modules/@ckeditor/ckeditor5-dev-docs/node_modules/docs-builder/node_modules/bootstrap-sass/assets/stylesheets/bootstrap
Parent style sheet: stdin
Details:
    formatted: Error: File to import not found or unreadable: /www/ckeditor5/ckeditor5/node_modules/@ckeditor/ckeditor5-dev-docs/node_modules/docs-builder/node_modules/bootstrap-sass/assets/stylesheets/bootstrap
       Parent style sheet: stdin
        on line 2 of stdin
>> @import '/www/ckeditor5/ckeditor5/node_modules/@ckeditor/ckeditor5-dev-docs/no
   ^

    column: 1
    line: 2
    file: stdin
    status: 1
Stack:
Error: File to import not found or unreadable: /www/ckeditor5/ckeditor5/node_modules/@ckeditor/ckeditor5-dev-docs/node_modules/docs-builder/node_modules/bootstrap-sass/assets/stylesheets/bootstrap
Parent style sheet: stdin
    at options.error (/www/ckeditor5/ckeditor5/node_modules/@ckeditor/ckeditor5-dev-docs/node_modules/node-sass/lib/index.js:292:32)
(t/243 daae923) p@m /www/ckeditor5/ckeditor5>
```

**Comment by Reinmar** [comment]
And:

```
(t/1 a7903db) p@m /www/ckeditor5/ckeditor5-dev-docs> ls node_modules/bootstrap-sass/
CHANGELOG.md		LICENSE			README.md		assets			eyeglass-exports.js	package.json
```

**Comment by Reinmar** [comment]
OK, my fault. I linked docs-builder in the wrong package.

So now I've got:

```
[11:12:44] Building the guides main page...
[11:12:44] An error occurs during the build process.
[11:12:44] 'docs:build' errored after 1.04 s
[11:12:44] Error: ENOENT: no such file or directory, open '/www/ckeditor5/ckeditor5/build/docs/guides/index.html'
    at Error (native)
```

**Comment by pomek** [comment]
Fixed. Now I can get back to work on README.
